gremlin-go/driver/traversal.go

```diff
@@ -477,3 +479,32 @@ var WithOptions withOptions = withOptions{
 	List:    0,
 	Map:     1,
 }
+
+type Metrics struct {
+	Id   string
+	Name string
+	// the duration in nanoseconds.
+	Duration      int64
+	Counts        map[string]int64
+	Annotations   map[string]interface{}
+	NestedMetrics []Metrics
+}
+
+type TraversalMetrics struct {
+	// the duration in nanoseconds.
+	Duration int64
+	Metrics  []Metrics
+}
+
+type GremlinType struct {
+	Fqcn string
+}
+
+type BigDecimal struct {
+	Scale         int32
+	UnscaledValue big.Int
+}
+
+type ByteBuffer struct {
```

Comment:
Add comments in front of public structs for documentation. Also struct above this - WithOptions could use a comment as well (my bad missing that one).